.ci/pitest.groovy

```diff
@@ -0,0 +1,459 @@
+ * are parsed.
+ *
+ * @param mainNode the main {@code suppressedMutations} node
+ * @return A set of suppressed mutations
+ */
+private static Set<Mutation> getSuppressedMutationList(Node mainNode) {
+    final def children = mainNode.children()
+    final Set<Mutation> suppressedMutations = new TreeSet<>()
+
+    children.each { node ->
+        final Node mutationNode = node as Node
+        suppressedMutations.add(getMutation(mutationNode))
+    }
+    return suppressedMutations
+}
+
+/**
+ * Construct the {@link Mutation} object from the {@code mutation} XML node.
+ *
+ * @param mutationNode the {@code mutation} XML node
+ * @return {@link Mutation} object represented by the {@code mutation} XML node
+ */
+private static Mutation getMutation(Node mutationNode) {
+    final List childNodes = mutationNode.children()
+
+    String sourceFile = null
+    String mutatedClass = null
+    String mutatedMethod = null
+    String mutator = null
+    String lineContent = null
+    String description = null
+    String mutationClassPackage = null
+    int lineNumber = 0
+    childNodes.each {
+        final Node childNode = it as Node
+        final String text = childNode.name()
+
+        final String childNodeText = XmlUtil.escapeXml(childNode.text())
+        switch (text) {
+            case "sourceFile":
+                sourceFile = childNodeText
+                break
+            case "mutatedClass":
+                mutatedClass = childNodeText
+                mutationClassPackage = mutatedClass.split("[A-Z]")[0]
+                break
+            case "mutatedMethod":
+                mutatedMethod = childNodeText
+                break
+            case "mutator":
+                mutator = childNodeText
+                break
+            case "description":
+                description = childNodeText
+                break
+            case "lineNumber":
+                lineNumber = Integer.parseInt(childNodeText)
+                break
+            case "lineContent":
+                lineContent = childNodeText
+                break
+        }
+    }
+    if (lineContent == null) {
+        final String mutationFileName = mutationClassPackage + sourceFile
+        final String startingPath = ".${SEPARATOR}src${SEPARATOR}main${SEPARATOR}java${SEPARATOR}"
+        final String javaExtension = ".java"
+        final String mutationFilePath = startingPath + mutationFileName
+                .substring(0, mutationFileName.length() - javaExtension.length())
+                .replaceAll("\\.", SEPARATOR) + javaExtension
+
+        final File file = new File(mutationFilePath)
+        lineContent = XmlUtil.escapeXml(file.readLines().get(lineNumber - 1).trim())
+    }
+    if (lineNumber == 0) {
+        lineNumber = -1
+    }
+    boolean unstable = false
+    final String unstableMutation = mutationNode.attribute("unstable")
+    if (unstableMutation != null) {
+        unstable = Boolean.parseBoolean(unstableMutation)
+    }
```

Comment:
Please remove condition, this method is null safe. `final boolean isUnstable`

Reply:
Done.

Reply:
Not done, `isUnstable`.

Reply:
Reason at https://github.com/checkstyle/checkstyle/pull/11731#discussion_r903735724 does not matter here :)

Reply:
Done now.